### PR TITLE
removed /* comments */ from sql templates during processing

### PR DESCRIPTION
**Pull request by ee-usgs**
Someday we may have optimizer hints in comments, but for now it seems like they just make the debug way harder to read.

### CODE REVIEW

File: src/main/java/gov/acwi/wqp/etl/SqlTemplateTasklet.java [hunk 1]
```diff
@@ -56,7 +55,9 @@ public String getSqlTemplate() {
 
 	public String getSqlString(Map<String, String> params) {
 
-		var wrap = new Object() { public String text = getSqlTemplate(); };
+		final String noComment = getSqlTemplate().replaceAll("(?s)\\/\\*.*?\\*\\/", ""); //remove all /* ... */ comments
```
Comment (orosz-usgs) [inline src/main/java/gov/acwi/wqp/etl/SqlTemplateTasklet.java:58]: / is not a special charcter in regular expression. Think it should be
replaceAll("(?s)/\\*.*?\\*/", "")

File: src/test/java/gov/acwi/wqp/etl/AttachRangePartitionTableTest.java [hunk 2]
```diff
@@ -14,9 +14,6 @@ public void getSqlStringTest() {
 
 		String str = task.getSqlString();
 
-		//Strip out the comments at the top of the file, since they will look the 'correct answer'.  Comments end with "*/".
-		str = str.substring(str.indexOf("*/") + 2);
-
 		assertTrue(str.matches("(?si).*alter table\\s+xyz\\.parenttable.*"));
```
Comment (orosz-usgs) [inline src/test/java/gov/acwi/wqp/etl/AttachRangePartitionTableTest.java:17]: the .* would still match a comment that follows parenttable

Comment (ee-usgs) [reply src/test/java/gov/acwi/wqp/etl/AttachRangePartitionTableTest.java:17]: This bit of code is being removed.  It was simpler code that was making an assumption that the comment was only at the start of the file and there was only a single comment.  The new regex is more generic and handles multiple comments.

File: src/main/java/gov/acwi/wqp/etl/SqlTemplateTasklet.java (hunk 1 above)
Comment (ee-usgs) [reply src/main/java/gov/acwi/wqp/etl/SqlTemplateTasklet.java]: Ah, yep.  I pulled out the escaping for the forward slash.
I don't think the other changes you propose are correct - for one, I at least need to make sure the string starts with '/*'.  Also, I'm not sure what '.?' would do - '?' is not a quantifier by itself I don't think, but in combination with '.*?' It should be any number of character, match non-greedily.

Comment (orosz-usgs) [reply src/main/java/gov/acwi/wqp/etl/SqlTemplateTasklet.java]: sorry, had typo

File: src/test/java/gov/acwi/wqp/etl/AttachRangePartitionTableTest.java (hunk 2 above)
Comment (orosz-usgs) [reply src/test/java/gov/acwi/wqp/etl/AttachRangePartitionTableTest.java:17]: missed the line below, it's good.

**Review by orosz-usgs — APPROVED** [review]

**Pull request merged**